antifmt.sh

```diff
@@ -16,6 +16,14 @@ if [ -z "$*" ]; then
 	exit 1
 fi
 
+for cmd in unzip unrar 7zr bunzip2 gunzip unxz tar pdftotext
+do
+	if ! command -v $cmd >/dev/null 2>&1
+	then
+		echo "Who am I? Why am I here? Am I on lilo? $cmd is missing!" >& 2
+		exit 1
+	fi
+done
```

Comment:
I would suggest factoring this out into a function provided by a `common.sh` which would be sourced by the other scripts.
eg.

```sh
validate_externals () {
  for cmd in $@
  do
    if ! command -v $cmd > /dev/null 2>&1
    then 
      echo "Error" >&2
      exit 1
    fi
  done
}
```

Later:

```sh
validate_externals unzip unrar
```

Reply:
Good, I was hesitant of adding a `common.sh` but if it's fine, sure.

Reply:
I'm not sure what @squell thinks of it, but I think it should be fine. If we make sure it only contains definitions, we should also be protected against people running `./common.sh` accidentally.

Reply:
I would also be hesitant in adding a "global" script; it sort of feels right that most the scripts are useful stand-alone.

Also, when written in this fashion:
```bash
set -e
for cmd in gzip rar bla foo baz; do test -x "$(which "$cmd")"; done
```
It's only one line of code in a script to add really meaningful errors messages; that doesn't need to be factored out.

(By the way, does `command -v` have any benefits over `which`?)

Reply:
`command` is POSIX, `which` is an external program and behaves differently over systems.
`hash` and `type` are valid alternatives for `command` if you are fine with using bashisms.

I followed the ---  reasonably looking --- advice from https://stackoverflow.com/a/677212/5462598

Reply:
As a final funny trivia. `which` on debian is a `POSIX` compliant `sh` script:)

Reply:
Maybe; I actually hope not -- I think the paradigm of parcelling out functionality into separate scripts should prevent that.

But in any case, this particular use case doesn't convince me that a `common.sh` is necessary: adding it itself introduces something that potentially can lead to errors (example of this: you forgot to distribute common.sh in `verdeel.sh`). That is a net negative for the intended audience in exchange for a more gracious failure mode for some (a few?) others.

That being said: **I like your proposal** to add safety belts. If you are only checking at most, say, three programs, I would simply manually instantiate $cmd in this single line of code:
```bash
command -v "$cmd" /dev/null || { echo "meaningful error about $cmd" 1>&2; exit 1; }
```
And then use a  loop for scripts like antifmt.sh that calls a bunch of programs, as you did previously. 

I don't think that's more repetitive (or likely to become a maintenance issue) than say, the command line help in every script.